roscala/src/main/scala/coop/rchain/roscala/ob/Tuple.scala

```diff
@@ -14,6 +15,101 @@ class Tuple(val value: Array[Ob]) extends Ob {
 
   def update(arg: Int, ob: Ob): Unit = value.update(arg, ob)
 
+  def accepts(msg: Ctxt): Boolean =
+    if (this == Nil) {
+      true
+    } else {
+      this.value.exists(_.matches(msg))
+    }
+
+  def makeSlice(offset: Int, n: Int): Tuple =
+    this match {
+      case Nil => Nil
+      case _   => Tuple(n, this, offset, n)
+    }
+
+  def makeTail(entriesToSkip: Int): Tuple = {
+    val size = numberOfElements() - entriesToSkip
+    Tuple(size, this, entriesToSkip, size)
+  }
+
+  override def matches(msg: Ctxt): Boolean = {
+    val n = numberOfElements()
+
+    if (n > 0 && n <= msg.argvec.value.length) {
+      if (this.value.head != msg.trgt) {
+        false
+      } else {
+
+        /*
+         * Skip msg.head, since it is actually the receiver of the
+         * message, i.e.,
+         *
+         * 	(oprn rcvr arg1 arg2 ...)
+         *
+         * results in an argvec of
+         *
+         * 	[rcvr arg1 arg2 ...]
+         */
+
+        this.value
+          .drop(1)
+          .zip(msg.argvec.value)
```

Comment:
You don't want to compare `this.value(1)` with `msg.argvec.value(0)` but that's what you do here.

Reply:
Yes, you are right.